students/elaine_x/session09/mailroom5_cli.py

```diff
@@ -0,0 +1,93 @@
+'''
+##########################
+#Python 210
+#Session 09 - Mailroom Part 5
+#Elaine Xu
+#Mar 13, 2019
+###########################
+'''
+import sys
+from mailroom5_donor_models import Donor
+from mailroom5_donor_models import DonorCollection
+
+#############################################################
+if __name__ == "__main__":
+    DIC_MENU = ('''
+                1: send_a_thankyou,
+                2: create_a_report,
+                3: exit_program'''
+                )
+
+    PROMPT = ('Menu\n'
+              'Please choose from below options:\n'
+              '1 - Send a Thank You\n'
+              '2 - Create a Report\n'
+              '3 - Quit\n'
+              'Enter your selection: ')
+
+    donor_db = {"William Gates, III": [653772.32, 2],
+                "Jeff Bezos": [877.33, 1],
+                "Paul Allen": [663.23, 3],
+                "Mark Zuckerberg": [1663.23, 3],
+                "Bob Smith": [500.00, 1],
+                }
+    donor_db_lower = [donor.lower() for donor in donor_db]
+
+    for donor in donor_db:
+        DonorCollection.add_donor(donor, donor_db[donor])
+
+    while True:
+        try:
+            choice = int(input(PROMPT))
+            if choice == 1:
+                while True:
+                    fullname = input("Enter full name of the donor (type 'list' to show all "
+                                     "donor names): ")
+                    if fullname.lower() == 'list':
+                        print(DonorCollection.list_all_names())
+                    elif fullname.lower() not in donor_db_lower:
```

Comment:
You should be searching in the donor collection. This causes that, if a new donor is added, any subsequent donation for the new donor will still report the donor as being new, causing the new donation to replace, instead of being added to, the original donation.